apps/src/code-studio/initApp/loadApp.js

```diff
@@ -315,46 +315,59 @@ async function loadAppAsync(appOptions) {
     }?section_id=${sectionId}`
   );
 
-  try {
-    const [data, exampleSolutions] = await Promise.all([
-      userAppOptionsRequest,
-      exampleSolutionsRequest
-    ]);
-
-    appOptions.exampleSolutions = exampleSolutions;
-    appOptions.disableSocialShare = data.disableSocialShare;
-
-    // We do not need to process data.progress here because labs do not use
-    // the level progress data directly. (The progress bubbles in the header
-    // of the level pages are rendered by header.build in header.js.)
-
-    if (data.lastAttempt) {
-      appOptions.level.lastAttempt = data.lastAttempt.source;
-    } else if (!data.signedIn) {
-      // User is not signed in, load last attempt from session storage.
-      appOptions.level.lastAttempt = clientState.sourceForLevel(
-        appOptions.scriptName,
-        appOptions.serverProjectLevelId || appOptions.serverLevelId
-      );
-    }
-
-    appOptions.level.isNavigator = data.isNavigator;
-    if (data.pairingDriver) {
-      appOptions.level.pairingDriver = data.pairingDriver;
-      appOptions.level.pairingAttempt = data.pairingAttempt;
-      appOptions.level.pairingChannelId = data.pairingChannelId;
-    }
+  const [
+    userAppOptionsResponse,
+    exampleSolutionsResponse
+  ] = await Promise.allSettled([
+    userAppOptionsRequest,
+    exampleSolutionsRequest
+  ]);
+
+  if (exampleSolutionsResponse.status === 'rejected') {
```

Comment:
(minor) I feel like this is somehow obscuring the logic a bit (possibly even more so than we just used a .then block).  Would it be simpler if we just had two separate await blocks instead of using Promise.allSettled?

Side note: It looks like the handling is currently the same for both requests -- if the requests fails, we silently fail but continue to load the page.  Long-term, I think we'll want to show an error to the user if userAppOptionsRequest fails (but maybe not if exampleSolutionsRequest fails), right?

Reply:
If we have two separate awaits then the requests will not run in parallel, each await will wait for the response before continuing. I can try refactoring to use .then, I generally prefer await because I think it's easier to read. 

The way I wrote this, if example solutions request fails, we'll do a console log and continue to execute. If app options fails I return the app options that we have, which is how it was handled before if the app options request failed. I agree we should probably re-think what we want to happen if that request fails

Reply:
Oh, I thought the requests get initiated (i.e. sent) when `$.ajax` is called and the `await` just waits for it to finish?  (In general, I agree that `await` is a more modern pattern and we should favor it. I was just surprised that for some reason this feels harder to read.)

Reply:
When you use `await` the next line isn't executed until the the request finishes.
```
const responseA = await RequestA();
const responseB = await RequestB();
doOtherStuff()
```
would be the same as
```
RequestA.then(responseA => {
  RequestB.then(responseB => {
    doOtherStuff()
  })
})
```
Let me know if that helps, also happy to do a call and discuss this more!

I think the reason this one is harder to read is because instead of getting a direct result from `allSettled` it returns objects with statuses and results which means we need to check another layer.

Reply:
Let me see if I can replace `allSettled` with something that's easier to work with

Reply:
Oh, I was thinking more along the lines of leaving everything before line 318 as-is and then replacing the rest of the function with something like:
```
try {
    exampleSolutions = await exampleSolutionsRequest
} catch (err) {
}
...
try {
    userAppOptions = await userAppOptionsRequest
} catch (err) {
}
```

Reply:
So if it was written like that, the example solutions request would be made and then after it returns and the stuff in between executes, then the request to appOptions would be made. The requests would not be made in parallel

Reply:
Yes, even if the result is assigned in a separate try block the execution is still blocked until the promise resolves

Reply:
Following up on this after a conversation James and I had in slack. These requests will run in parallel if ajax is called for each before the await statements, I've updated the PR